agent/agent.go

```diff
@@ -539,16 +550,18 @@ func (a *Agent) reloadWatches(cfg *config.RuntimeConfig) error {
 		watchPlans = append(watchPlans, wp)
 	}
 
-	// Stop the current watches.
-	for _, wp := range a.watchPlans {
-		wp.Stop()
-	}
-	a.watchPlans = nil
-
-	// deterine the primary http endpoint
-	addr := cfg.HTTPAddrs[0].String()
-	if cfg.HTTPAddrs[0].Network() == "unix" {
-		addr = "unix://" + addr
+	// Determine the primary http(s) endpoint.
+	var addr string
+	if len(cfg.HTTPAddrs) > 0 {
+		addr = cfg.HTTPAddrs[0].String()
+		if cfg.HTTPAddrs[0].Network() == "unix" {
+			addr = "unix://" + addr
+		}
+	} else {
+		addr = cfg.HTTPSAddrs[0].String()
```

Comment:
This assumes that if there is no httpADDrs set then HttpsAddrs will be set. Shouldn't that be ``` else if len(cfg.HttpsAddrs) >0) { ...```

Reply:
@preetapan There is an earlier check that one of them must be not empty.

Reply:
Maybe you can write it like this. Not sure if this is much better though:

```go
var netaddr net.Addr
if len(cfg.HTTPAddrs) > 0 {
    netaddr = cfg.HTTPAddrs[0]
} else {
    netaddr = cfg.HTTPSAddrs[0]
}
addr := netaddr.String()
if netaddr.Network() == "unix" {
    addr = "unix://" + addr
}
```